src/Persistence/YamlValidator/IValidatorFactory.cs

```diff
@@ -0,0 +1,9 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.PowerPlatform.PowerApps.Persistence.YamlValidator;
+
+public interface IValidatorFactory
+{
+    IValidator GetValidator();
```

Comment:
'Get' conveys that the thing being retrieved already exists. member data or db data lookup.
'Create' conveys a new instance will always be returned.
There are some exceptions, but from the consumer of the interface, which is a Factory pattern, the expectation is that a new instance should get constructed. So rename to 'CreateValidator'.